eng/pipelines/maui-release.yml

```diff
@@ -190,4 +196,85 @@ extends:
+                    Write-Host "##vso[task.setendpoint id=$(MauiNightlyFeedServiceConnectionId);field=authParameter;key=apitoken]${accessToken}"
+
+              - task: NuGetAuthenticate@0
+                displayName: NuGet Authenticate using managed identity token
+                inputs:
+                  nuGetServiceConnections: maui-nightly-feed
+
+              - task: DownloadPipelineArtifact@2
+                inputs:
+                  artifactName: DropMetadata-$(Build.BuildId)-nugets-$(System.JobAttempt)
+                  downloadPath: $(Build.StagingDirectory)\metadata
+                displayName: Download nugets drop metadata
+
+              - powershell: |
+                  $jsonContent = Get-Content -Path "$(Build.StagingDirectory)\metadata\VSTSDrop.json" -Raw | ConvertFrom-Json
+                  $dropPrefix = $jsonContent.VstsDropBuildArtifact.VstsDropUrl -replace 'https://devdiv.artifacts.visualstudio.com/DefaultCollection/_apis/drop/drops/' -replace '/nugets'
+                  Write-Host "##vso[task.setvariable variable=ReleaseDropPrefix]$dropPrefix"
+                displayName: Set variable ReleaseDropPrefix
+
+              - task: ms-vscs-artifact.build-tasks.artifactDropDownloadTask-1.artifactDropDownloadTask@1
+                displayName: Download $(ReleaseDropPrefix)/nugets
+                inputs:
+                  dropServiceURI: https://devdiv.artifacts.visualstudio.com/DefaultCollection
+                  buildNumber: $(ReleaseDropPrefix)/nugets
+                  destinationPath: $(Build.StagingDirectory)\nuget-signed
+
+              - powershell: |
+                  $feed = "$env:NUGET_FEED_MAUI_NIGHTLY"
+                  # Signed maui packages except manifest
+                  $nupkgs = (Get-ChildItem -Path "$(Build.StagingDirectory)\nuget-signed\" -Filter *.nupkg) | Where-Object { $_.BaseName -notlike 'Microsoft.NET.Sdk.Maui.Manifest-*' }
+                  $maxAttempts = 5
+                  foreach($nupkg in $nupkgs) {
+                      $nupkgFile = $nupkg.FullName
+                      $attempt = 1
+                      $waiting = $true
+                      do {
+                          try {
+                              Write-Output "dotnet push $nupkgFile"
+                              & dotnet nuget push --source $feed --api-key az --skip-duplicate $nupkgFile
+                              $waiting = $false
+                          }
+                          catch {
+                              if ($attempt -gt $maxAttempts)
+                              {
+                                throw 'Maximum attempts reached, failing!'
+                              }
+                              Write-Output "  attempt $attempt of $maxAttempts failed..."
+                              $attempt = $attempt + 1
```

Comment:
How about a sleep here with incremental backoff?

Reply:
I think this is just a copy of what we have in the release, we can improve, but this seems to be working for us till now :)